datanode/src/main/java/com/odysseusinc/arachne/datanode/model/analysis/AnalysisStateEntry.java

```diff
@@ -39,74 +42,15 @@ public class AnalysisStateEntry {
     private Long id;
     @Column(name = "date")
     private Date date;
-    @Column(name = "state")
```

Comment:
Major: Removing state here will break histories of existing analyses that didn't have 'stage'
I suggest we stay a bit more conservative with DB fields:
in AnalysisStateEntry, keep everything as it is, only adding stage and error.
in Analysis entity, keep everything as it is and _only_ add reference to current state entry and start time,
Yes, with this approach stage and error will be both in AnalysisStateEntry and Analysis, however that exact fact makes it much easier to deal with potential migration issues. And it's not difficult to maintain, since there should be essentially 1-2 places where stage is written.

Reply:
The old analyses will stay in their current status as they are now (they’re not running and won’t progress further, so no new records will be added to the `analysis_state_journal`). We just need the final status for them. I don’t see any current use for the execution history. That status was moved to `analysis_state_journal`, so stages aren’t really necessary for old analyses.

Additionally, the `status` field hasn’t been removed from the database - we can always add a migration later if needed. I’d rather focus on testing for now, and if any issues come up, the information is still available in the `analysis_state_journal`.